tests/test_kernel_language.py

```diff
@@ -150,3 +150,29 @@ class TestParticle(ptype[mode]):
                          'random.%s(%s)' % (rngfunc, ', '.join([str(a) for a in rngargs])))
     pset.execute(kernel, endtime=1., dt=1.)
     assert np.allclose(np.array([p.p for p in pset]), series, rtol=1e-12)
+
+
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+def test_c_kernel(fieldset, mode):
+    pset = ParticleSet(fieldset, pclass=ptype[mode], lon=[0.5], lat=[0])
+
+    def c_func(U):
+        print('Calling the python func')
+        print('%g %g' % (U.data[0, 0, 0], U.data[0, 0, 1]))
+        print('%g %g' % (U.data[0, 1, 0], U.data[0, 1, 1]))
+
+    c_func_lib = """
+             static inline void c_func(CField *f)
+             {
+               printf("Calling the C func\\n");
+               float (*data)[f->xdim] = (float (*)[f->xdim]) f->data;
+               printf("%g %g\\n", data[0][0], data[0][1]);
+               printf("%g %g\\n", data[1][0], data[1][1]);
```

Comment:
Nice, but would it not make more sense to have a kernel that actually *does* something? I.e. not only print, but also update the particle data? Then we can also add an `assert` to check for more than just whether compilation succeeds